Added methods

woolgathering opened the pull request:
I added a few methods for convenience and new functionality.

__zero__: a method to easily create zero vectors of arbitrary size
__clip__: clip a vector (distinct from limit)
__isVector__: returns true if it's a Vector (AbstractVector)
__rotate__: part of RealVector2D to rotate a vector

woolgathering committed:
Added *zero method to create zero vectors easily.

woolgathering committed:
Added clip(), isVector, and zero

woolgathering committed:
rotate (in 2D) method added

woolgathering committed:
return false if not a Vector

woolgathering committed:
zero convenience method for RealVector2D

woolgathering committed:
Removed non UTF-8 character

woolgathering committed:
Update RealVector3D methods

woolgathering committed:
Cleanup and azi() method for RealVector3D

woolgathering commented:
Just another ping... These changes are needed for [Boids](https://github.com/woolgathering/boids).

nhthn commented:
dunno if this library is being actively maintained. cc @miguel-negrao

miguel-negrao commented:
The commit has a lot of whitespace changes, what was the reason for this ? Difficult to see what was changed...

woolgathering commented:
The earliest commit has a lot of whitespace because I didn't realize that the original class file used tab characters instead of spaces. I couldn't see it in my editor but saw it when I uploaded it to Github. Changed it back to tabs to be consistent later. This is the case with a lot of the whitespace (except some of the spaces just beyond the first curly braces; Atom removed them automatically on save...).

There were also some non-UTF8 characters that I deleted. Maybe I should have left those?

The actual changes:
- **zero()**: [RealVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L118), [RealVector2D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L195), and [RealVector3D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L228).
- **clip()**: [AbstractVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L38)
- **isVector()**: [AbstractVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L95)
- **rotate()**: [RealVector2D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L211)
- **azi()**: [RealVector3D](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L253)
- **asRealVector3D**: [AbstractVector](https://github.com/woolgathering/VectorSpace/blob/master/Vector.sc#L62)

Thoughts?

miguel-negrao commented on Vector.sc:
```diff
@@ -1,92 +1,106 @@
 		if(this.norm > max, { ^this.normalize * max })
 	}
-	
+
```
Indentation seems wrong here. Possibly not using tabs ?

woolgathering replied:
This is caused by the removal of trailing whitespace. Same thing happens all over the file.

miguel-negrao commented on Vector.sc:
```diff
@@ -1,92 +1,106 @@
+		}
 	}
-		
+
```
indentation
See comment at + Object

miguel-negrao commented on Vector.sc:
```diff
@@ -213,16 +250,20 @@ RealVector3D[slot] : RealVector {
 
 	phi { ^atan2(this[2], (this[0].squared + this[1].squared).sqrt) }
 
+	azi {
```
Shouldn't azimuth be arctan( y / x ) ?

See [wikipedia](https://en.wikipedia.org/wiki/Spherical_coordinate_system#Coordinate_system_conversions)

woolgathering replied:
Yes, I can do it that way to keep things consistent. I just through of turning the entire 3D space on its side and doing the same operation about the x-axis as opposed to the z-axis.

miguel-negrao commented on object.sc:
```diff
@@ -0,0 +1,6 @@
++ Object {
```
You can do o.isKindOf(AbstractVector). Is this really needed ?

woolgathering replied:
Not anymore. I removed that line in Boids and can remove it here, too.

miguel-negrao commented on Vector.sc:
```diff
@@ -1,92 +1,106 @@
-// �2009 Miguel Negr�o, Fredrik Olofsson
+// 2009 Miguel Negro, Fredrik Olofsson
```
Please correct to 

    Copyright 2009 Miguel Negrão, Fredrik Olofsson

woolgathering replied:
No problem.

miguel-negrao commented on Fields.sc:
```diff
@@ -1,157 +1,157 @@
-// �2009 Miguel Negr�o
+// 2009 Miguel Negro
```
Please change to "Copyright 2009 Miguel Negrão"

woolgathering replied:
See above

miguel-negrao reviewed (changes requested):
Regarding the whitespace changes, trailing whitespace is fine (thanks !) but changes from tabs to spaces is not.

miguel-negrao commented:
If your git skills are up to it, it would be nice to have a single commit just with the trailing whitespace and weird character removal changes, followed by a single commit with your actual changes.

woolgathering commented:
No problem. I'll get to it this weekend!

woolgathering commented:
I'll close this request and make a new one that's cleaner.

miguel-negrao commented:
sorry, forgot the word "removing" in "Regarding the whitespace changes, REMOVING trailing whitespace is fine (thanks !) but changes from tabs to spaces is not."
